examples/baselines/bc/bc_rgbd.py

```diff
@@ -0,0 +1,510 @@
+            sim_backend="cpu",
+        )
+        if video:
+            x = RecordEpisode(
+                x, os.path.join(log_path, args.video_path), save_trajectory=False, info_on_video=True
+            )
+        x = gym.wrappers.ClipAction(x)
+        x = CPUGymWrapper(x)
+
+        x.action_space.seed(seed)
+        x.observation_space.seed(seed)
+        return x
+
+    return inner
+
+
+def make_eval_envs(env, num_envs, control_mode,gpu= False):
+    if gpu:
+        env = gym.make(
+            env,
+            num_envs=num_envs,
+            obs_mode="rgbd",
+            control_mode=control_mode,
+            render_mode="sensors",
+        )
+        env = ManiSkillVectorEnv(env)
+        if args.video:
+            env = RecordEpisode(
+                env, os.path.join(log_path, args.video_path), save_trajectory=False, max_steps_per_video= args.max_traj_len
+            )
+        return env
+
+    return gym.vector.SyncVectorEnv(
+        [
+            build_env(
+                env,
+                video=(j == 0 and args.video),
+                seed=j,
+                control_mode = control_mode,
+            )
+            for j in range(num_envs)
+        ]
+    )
+
+def process_obs_gpu(obs):
+    out = {}
+    out["rgb"] = obs["sensor_data"]["base_camera"]["rgb"]
+    out["depth"] = obs["sensor_data"]["base_camera"]["depth"]
+    out["state"] = torch.hstack([
+        flatten_state_dict_with_space(obs["agent"]),
+        flatten_state_dict_with_space(obs["extra"])
+    ])
+    return out
+
+def process_obs(obs, device):
+    out = {}
+    out["rgb"] = torch.from_numpy(obs["sensor_data"]["base_camera"]["rgb"]).to(device=device)
+    out["depth"] = torch.from_numpy(obs["sensor_data"]["base_camera"]["depth"]).to(device=device)
+    out["state"] = torch.from_numpy(np.hstack([
+        flatten_state_dict_with_space(obs["agent"]),
+        flatten_state_dict_with_space(obs["extra"])
+    ])).to(device=device)
+    return out
+
+def save_ckpt(tag):
+    os.makedirs(f'{log_path}/checkpoints', exist_ok=True)
+    torch.save({
+        'actor': actor.state_dict(),
+    }, f'{log_path}/checkpoints/{tag}.pt')
+
+
+if __name__ == "__main__":
+    args = tyro.cli(Config)
+
+    now = datetime.datetime.now().strftime("%y%m%d-%H%M%S")
+    tag = '{:s}_{:d}'.format(now, args.seed)
+    log_name = os.path.join(args.env, "BC_RGB", tag)
+    log_path = os.path.join(args.output_dir, log_name)
+
+    device = torch.device(args.device if torch.cuda.is_available() else "cpu")
+    control_mode = os.path.split(args.demo_path)[1].split(".")[2]
+
+
+    if args.wandb:
+        wandb_init(asdict(args))
+
+    ds = ManiSkillDataset(
+        args.demo_path,
+        device=device,
+        load_count=args.load_count,
+    )
+
+    gpu = args.sim_backend == "gpu"
+
+    if args.video:
+        os.makedirs(os.path.join(log_path, args.video_path))
+
+
+    envs = make_eval_envs(args.env, args.num_envs, control_mode, gpu)
```

Comment:
I recommend sticking to using the flatteng RGBD observation wrapper as used in the PPO rgb code instead of making all the extra process obs functions.